### PR TITLE
Fix #488 in fortran output

### LINKED ISSUE #488: dr/cpp output
**Describe the bug**
The output of dr/cpp branch, does not fit the master branch. Only the quantity `Ts0` is different. Therefore, I suspect that the computations are done correctly, we just miss the nucleation stress in the output.

**Expected behavior**
Output of dr/cpp branc and master branch should be the same.

**To Reproduce**
1. master: e4118559 (not so current, but latest master should not make a difference) dr/cpp: e4c45ff0
2. Which build settings do you use? Which compiler version do you use? 
Not relevant
3. On which machine does your problem occur? If on a cluster: Which modules are loaded?
Not relevant
4. Provide parameter/material files.
tpv104 from https://github.com/7bits-register/precomputed-seissol

**Screenshots/Console output**
dr/cpp: `Ts0` after 2s
dr-cpp
master: `Ts0` after 2s
master

**Additional context**
As the problem occurs only for `Ts0` and not `T_s` or `Td0`b, I suspect that the nucleation stress is the problem. Also the diff of both outputs, has the same shape as the nucleation function. Note: the relative error is too small to be detected by the pipeline.

### PR DESCRIPTION
**Pull request by sebwolf-de**
This commit fixes issue https://github.com/SeisSol/SeisSol/issues/488, using the fortran output. Note that in the cpp implementation, we accumulate the initial stress and the nucleationstress together. Still have to check the cpp output.

### CODE REVIEW

File: src/ResultWriter/faultoutput.f90 [hunk 1]
```diff
@@ -555,7 +555,7 @@ SUBROUTINE calc_FaultOutput( DynRup_output, DISC, EQN, MESH, MaterialVal, BND, t
 
               !add transient nucleation stress to fault output
               IF ((EQN%FL.EQ.103) .OR. (EQN%FL.EQ.3) .OR. (EQN%FL.EQ.4)) THEN
```
Comment (Thomas-Ulrich) [inline src/ResultWriter/faultoutput.f90:557]: Then if I understand correctly, you need to remove this if statement?
Else you won't see the initial stress on the fault output with other friction laws?

Comment (sebwolf-de) [reply src/ResultWriter/faultoutput.f90]: You're right. I've removed it and checked tpv5 and 104, both work.

File: src/ResultWriter/faultoutput.f90 [hunk 2]
```diff
@@ -554,21 +554,17 @@ SUBROUTINE calc_FaultOutput( DynRup_output, DISC, EQN, MESH, MaterialVal, BND, t
...
-                 Tnuc = DISC%DynRup%t_0
-                 Gnuc = Calc_SmoothStep(time, Tnuc)
-                 NucleationStressLocalCS = MATMUL(rotmat, NucleationStressXYZ)*Gnuc
-              ELSE
-                 NucleationStressLocalCS(:) = 0d0
-              ENDIF
+              NucleationStressXYZ = MATMUL(T(1:6,1:6), EQN%InitialStressInFaultCS(iBndGP,:,iFace))
+              Tnuc = DISC%DynRup%t_0
+              Gnuc = Calc_SmoothStep(time, Tnuc)
+              NucleationStressLocalCS = MATMUL(rotmat, NucleationStressXYZ)*Gnuc
```
Comment (Thomas-Ulrich) [inline src/ResultWriter/faultoutput.f90:560]: hum, I m not sure this works.
stress = transient + initial stress + Gnuc* nucleation_overstress.
but not stress = transient + Gnuc* (initial stress + nucleation_overstress).
So maybe, the idea of merging initial stress and nucleation overstress does not work?

File: src/DynamicRupture/Output/Base.hpp [hunk 3]
```diff
@@ -28,16 +28,22 @@ class Base {
     }
 
     bool bothEnabled = generalParams.outputPointType == OutputType::AtPickpointAndElementwise;
-    if (generalParams.outputPointType == OutputType::AtPickpoint || bothEnabled) {
+    bool pointEnabled = generalParams.outputPointType == OutputType::AtPickpoint || bothEnabled;
+    bool elementwiseEnabled = generalParams.outputPointType == OutputType::Elementwise || bothEnabled;
+    if (pointEnabled) {
+      logInfo() << "Enable Dynamic Rupture Receiver Output";
```
Comment (Thomas-Ulrich) [inline src/DynamicRupture/Output/Base.hpp:34]: Enabling on-fault receivers output?

File: src/DynamicRupture/Output/Base.hpp [hunk 4]
```diff
@@ -28,16 +28,22 @@ class Base {
...
       ppOutputBuilder = std::make_unique<PickPointBuilder>();
       ppOutputBuilder->setMeshReader(&mesher);
       ppOutputBuilder->setParams(reader.getPickPointParams());
-    } else if (generalParams.outputPointType == OutputType::Elementwise || bothEnabled) {
+    }
+    if (elementwiseEnabled) {
+      logInfo() << "Enable Dynamic Rupture 2D Fault Output";
```
Comment (Thomas-Ulrich) [inline src/DynamicRupture/Output/Base.hpp:40]: Enabling 3D fault output?

File: src/DynamicRupture/Output/Base.hpp [hunk 5]
```diff
@@ -28,16 +28,22 @@ class Base {
...
       ewOutputBuilder = std::make_unique<ElementWiseBuilder>();
       ewOutputBuilder->setMeshReader(&mesher);
       ewOutputBuilder->setParams(reader.getElementwiseFaultParams());
-    } else {
-      logError() << "Unknown fault output type (not 3,4,5)";
+    }
+    if (!elementwiseEnabled && !pointEnabled) {
+      logInfo() << "No Dynamic Rupture Output enabled";
```
Comment (Thomas-Ulrich) [inline src/DynamicRupture/Output/Base.hpp:46]: remove capitalization.

File: src/ResultWriter/faultoutput.f90 (hunk 2 above)
Comment (sebwolf-de) [reply src/ResultWriter/faultoutput.f90]: This is how we do it: https://github.com/SeisSol/SeisSol/blob/dr/cpp/src/DynamicRupture/FrictionLaws/RateAndState.h#L146-L155, so with the Gnuc removed from this code  snippet, I think we are doing it correctly.

File: src/DynamicRupture/Output/Base.hpp (hunk 4 above)
Comment (sebwolf-de) [reply src/DynamicRupture/Output/Base.hpp:40]: I would have said 2D since the faults are 2D manifolds (neglecting intersections) embedded in 3D space.

**Review by Thomas-Ulrich — APPROVED** [review]
ok, I think it should works!

**Pull request merged**